### DISCUSSION TITLE
confused about Get() method for `optional & nullable`generic wrapper

### DISCUSSION THREAD

**Discussion by idledaemon** [ogen-go/ogen#961] [Q&A] [answered]
For example, if o.Null is true, should we return `v, true`?

https://github.com/ogen-go/ogen/blob/7a4f230e80c90b3997328cd565661ca7fc33137c/gen/_template/schema/generic.tmpl#L75C13-L75C18

```
// Get returns value and boolean that denotes whether value was set.
func (o OptNilString) Get() (v string, ok bool) {
	if o.Null {
		return v, false
	}
	if !o.Set {
		return v, false
	}
	return o.Value, true
}
```

**Comment by ernado** [comment, marked as answer]
The main reason of this is following semantic:
```go
if v, ok := foo.Value.Get(); ok {
    println("Value is set to v")
}
```
To save this semantic, v should be able to represent null value (e.g. `string | null`), othervise we can't distinguish bettween blank string and null.

This is not possible in Go, because we have no sum types.
